### ISSUE TITLE
Pivot table Field Settings have no effect

### ISSUE THREAD

**Issue by ghronkrepps** [ClosedXML/ClosedXML#613]
**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

0.90, 0.91 beta

**What is the current behavior?**

Pivot table Field Settings have no effect.

RepeatItemLabels
InsertBlankLines
ShowBlankItems
SubtotalsAtTop
Collapsed
IncludeNewItemsInFilter
InsertPageBreaks

**What is the expected behavior or new feature?**

Pivot table Field Settings should toggle the options on and off.

**Did this work in previous versions of our tool?  Which versions?**

No

**Code to reproduce problem:**
```c#

using System;

using ClosedXML.Excel;

class Program
{
        static void Main(string[] args)
        {            
            ClosedXMLPivotExample ex = new ClosedXMLPivotExample();

            ex.Create(@"C:\ClosedXML\PastryPivot.xlsx");

        }
}

```

```c#

using System;
using System.Collections.Generic;
using ClosedXML.Excel;

    class ClosedXMLPivotExample
    {
        private class Pastry
        {
            public Pastry(string name, int numberOfOrders, double quality, string month, DateTime bakeDate)
            {
                Name = name;
                NumberOfOrders = numberOfOrders;
                Quality = quality;
                Month = month;
                BakeDate = bakeDate;
            }

            public string Name { get; set; }
            public int NumberOfOrders { get; set; }
            public double Quality { get; set; }
            public string Month { get; set; }
            public DateTime BakeDate { get; set; }
        }

        public void Create(String filePath)
        {
            var pastries = new List<Pastry>
            {
                new Pastry("Croissant", 150, 60.2, "Apr", new DateTime(2016, 04, 21)),
                new Pastry("Croissant", 250, 50.42, "May", new DateTime(2016, 05, 03)),
                new Pastry("Croissant", 134, 22.12, "Jun", new DateTime(2016, 06, 24)),
                new Pastry("Doughnut", 250, 89.99, "Apr", new DateTime(2017, 04, 23)),
                new Pastry("Doughnut", 225, 70, "May", new DateTime(2016, 05, 24)),
                new Pastry("Doughnut", 210, 75.33, "Jun", new DateTime(2016, 06, 02)),
                new Pastry("Bearclaw", 134, 10.24, "Apr", new DateTime(2016, 04, 27)),
                new Pastry("Bearclaw", 184, 33.33, "May", new DateTime(2016, 05, 20)),
                new Pastry("Bearclaw", 124, 25, "Jun", new DateTime(2017, 06, 05)),
                new Pastry("Danish", 394, -20.24, "Apr", new DateTime(2017, 04, 24)),
                new Pastry("Danish", 190, 60, "May", new DateTime(2017, 05, 08)),
                new Pastry("Danish", 221, 24.76, "Jun", new DateTime(2016, 06, 21)),

                // Deliberately add different casings of same string to ensure pivot table doesn't duplicate it.
                new Pastry("Scone", 135, 0, "Apr", new DateTime(2017, 04, 22)),
                new Pastry("SconE", 122, 5.19, "May", new DateTime(2017, 05, 03)),
                new Pastry("SCONE", 243, 44.2, "Jun", new DateTime(2017, 06, 14)),
            };

            using (var wb = new XLWorkbook())
            {
                var sheet = wb.Worksheets.Add("PastrySalesData");
                // Insert our list of pastry data into the "PastrySalesData" sheet at cell 1,1
                var source = sheet.Cell(1, 1).InsertTable(pastries, "PastrySalesData", true);
                sheet.Columns().AdjustToContents();

                // Create a range that includes our table, including the header row
                var range = source.DataRange;
                var header = sheet.Range(1, 1, 1, 3);
                var dataRange = sheet.Range(header.FirstCell(), range.LastCell());

                IXLWorksheet ptSheet;
                IXLPivotTable pt;

                ptSheet = wb.Worksheets.Add("pvtFilter");

                pt = ptSheet.PivotTables.AddNew("pvtFilter", ptSheet.Cell(1, 1), dataRange);

                IXLPivotField pf = pt.RowLabels.Add("Month");

                pf.SetRepeatItemLabels(true);
                pf.SetShowBlankItems(true);
                pf.SetInsertBlankLines(true);
                pf.SetInsertPageBreaks(true);
                pf.SetSubtotalsAtTop(false);
                pf.SetCollapsed(false);
                pf.SetIncludeNewItemsInFilter(true);     

                pt.Values.Add("NumberOfOrders").SetSummaryFormula(XLPivotSummary.Sum);
                
                wb.SaveAs(filePath);
            }
        }
    }

```

Above code results in a spreadsheet with the field options that look like this:

![image](https://user-images.githubusercontent.com/5125907/33500003-40cc47aa-d6a5-11e7-997e-450f31835d31.png)

![image](https://user-images.githubusercontent.com/5125907/33499885-d15c0770-d6a4-11e7-9864-c947c9aa9ed5.png)

**Comment by igitur** [comment]
Thanks. Pivot tables are still work in progress. @b0bi79 has made some good work recently. See the pull requests. Maybe some of them address the issues that you listed.

**Comment by ghronkrepps** [comment]
I'm writing a system to build spreadsheets using Closed XML from a simple XML file and going through and implementing each Excel/Closed XML feature for pivot tables at the moment. Testing with the 0.91 beta branch and reporting any issues I come across.

I'll check out those pull requests.

**Comment by b0bi79** [comment]
I've fixed options for PivotTable, and ghronkrepps works with the PivotFields options. Unfortunately, I have not done anything for this yet.

**Comment by igitur** [comment]
@ghronkrepps As you can see, there's lots of work left. Sounds like you're a developer too, so please feel free to submit pull requests to improve the pivot tables.

**Comment by b0bi79** [comment]
Of course. I will work more :) 02.12.2017, 09:18, "Francois Botha" <notifications@github.com>:@ghronkrepps As you can see, there's lots of work left. Sounds like you're a developer too, so please feel free to submit pull requests to improve the pivot tables.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by igitur** [comment]
I was actually asking @ghronkrepps to contribute, but anybody welcome of course.

**Comment by b0bi79** [comment]
When I have time, I will deal with this issue. 02.12.2017, 09:26, "Francois Botha" <notifications@github.com>:I was actually asking @ghronkrepps to contribute, but anybody welcome of course.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.   -----------------С уважением, Алексей.

**Comment by b0bi79** [comment]
RepeatItemLabels. I did not find such a field in [pivotFieldDefinition](http://www.datypic.com/sc/ooxml/e-ssml_pivotField-1.html). What does it correspond to?

**Comment by b0bi79** [comment]
@igitur RepeatItemLabels. I did not find such a field in pivotFieldDefinition. What does it correspond to?

**Comment by ghronkrepps** [comment]
I did a comparison of two excel files, one with "repeat item labels" on and one with it off and here is the only difference in the file.

with option off
`
  <pivotField axis="axisCol" subtotalTop="0" showAll="0">
   <items count="4">
    <item x="1"/>
    <item x="0"/>
    <item x="2"/>
    <item t="default"/>
   </items>
  </pivotField>
`

with option on
`
  <pivotField axis="axisCol" subtotalTop="0" showAll="0">
   <items count="4">
    <item x="1"/>
    <item x="0"/>
    <item x="2"/>
    <item t="default"/>
   </items>
   <extLst>
    <ext xmlns:x14="http://schemas.microsoft.com/office/spreadsheetml/2009/9/main" uri="{2946ED86-A175-432a-8AC1-64E0C546D7DE}">
     <x14:pivotField fillDownLabels="1"/>
    </ext>
   </extLst>
  </pivotField>
`

**Comment by b0bi79** [comment]
Thanks. It turns out this option is Excel 2010, I have installed 2007.

**Comment by igitur** [comment]
Fixed in #616
